File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/FootballPlayerStats.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "football_player_stats", schema = "public")
public class FootballPlayerStats {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "football_player_stats_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "game_id")
    private Game game;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "player_id")
    private Player player;

    @Column(name = "total_tackles")
    private Long totalTackles;

    @Column(name = "solo_tackles")
    private Long soloTackles;

    @Column(name = "total_sacks")
    private Long totalSacks;

    @Column(name = "tackles_for_loss")
    private Long tacklesForLoss;

    @Column(name = "passes_defended")
    private Long passesDefended;

    @Column(name = "quarterback_hits")
    private Long quarterbackHits;

    @Column(name = "defensive_touchdowns")
    private Long defensiveTouchdowns;

    @Column(name = "total_fumbles")
    private Long totalFumbles;

    @Column(name = "fumbles_lost")
    private Long fumblesLost;

    @Column(name = "fumbles_recovered")
    private Long fumblesRecovered;

    @Column(name = "interceptions")
    private Long interceptions;

    @Column(name = "interception_yards")
    private Long interceptionYards;

    @Column(name = "interception_touchdowns")
    private Long interceptionTouchdowns;

    @Column(name = "number_of_field_goals")
    private Long numberOfFieldGoals;

    @Column(name = "field_goal_percentage")
    private Long fieldGoalPercentage;

    @Column(name = "longest_field_goal")
    private Long longestFieldGoal;

    @Column(name = "extra_points")
    private Long extraPoints;

    @Column(name = "extra_point_attempts")
    private Long extraPointAttempts;

    @Column(name = "points")
    private Long points;

    @Column(name = "number_of_kick_returns")
    private Long numberOfKickReturns;

    @Column(name = "total_kick_return_yards")
    private Long totalKickReturnYards;

    @Column(name = "number_of_yards_per_kick_return")
    private Double numberOfYardsPerKickReturn;

    @Column(name = "longest_kick_return_yards")
    private Long longestKickReturnYards;

    @Column(name = "kick_return_touchdowns")
    private Long kickReturnTouchdowns;

    @Column(name = "completions")
    private Long completions;

    @Column(name = "passing_attempts")
    private Long passingAttempts;

    @Column(name = "passing_yards")
    private Long passingYards;

    @Column(name = "yards_per_pass_attempt")
    private Double yardsPerPassAttempt;

    @Column(name = "passing_touchdowns")
    private Long passingTouchdowns;

    @Column(name = "interceptions_thrown")
    private Long interceptionsThrown;

    @Column(name = "number_of_times_sacked")
    private Long numberOfTimesSacked;

    @Column(name = "sacked_yards_lost")
    private Long sackedYardsLost;

    @Column(name = "passer_rating")
    private Double passerRating;

    @Column(name = "adjusted_qbr")
    private Double adjustedQbr;

    @Column(name = "number_of_punts")
    private Long numberOfPunts;

    @Column(name = "total_punt_yards")
    private Long totalPuntYards;

    @Column(name = "punt_yards_per_punt")
    private Double puntYardsPerPunt;

    @Column(name = "touchbacks")
    private Long touchbacks;

    @Column(name = "in20")
    private Long in20;

    @Column(name = "longest_punt_yards")
    private Long longestPuntYards;

    @Column(name = "number_of_punt_returns")
    private Long numberOfPuntReturns;

    @Column(name = "total_punt_return_yards")
    private Long totalPuntReturnYards;

    @Column(name = "punt_return_yards_per_punt_return")
    private Double puntReturnYardsPerPuntReturn;

    @Column(name = "longest_punt_return_yards")
    private Long longestPuntReturnYards;

    @Column(name = "punt_return_touchdowns")
    private Long puntReturnTouchdowns;

    @Column(name = "total_receptions")
    private Long totalReceptions;

    @Column(name = "total_receiving_yards")
    private Long totalReceivingYards;

    @Column(name = "average_yards_per_reception")
    private Double averageYardsPerReception;

    @Column(name = "receiving_touchdowns")
    private Long receivingTouchdowns;

    @Column(name = "longest_reception")
    private Long longestReception;

    @Column(name = "receiving_targets")
    private Long receivingTargets;

    @Column(name = "rushing_attempts")
    private Long rushingAttempts;

    @Column(name = "averages")
    private Double averages;

    @Column(name = "rushing_touchdowns")
    private Long rushingTouchdowns;

    @Column(name = "longest_run")
    private Long longestRun;

}

File: umbrella-insurance-react-ui/src/models/stats/playerStats/footballPlayerStats/v1/FootballPlayerStats.ts
import { Game } from "../../../../games/v1/Game";
import { Player } from "../../../../people/players/v1/Player";

export class FootballPlayerStats {
    id?: number;
    game?: Game;
    player?: Player;
    //Defense
    totalTackles?: number;//TOT
    soloTackles?: number;//SOLO
    totalSacks?: number;//SACKS
    tacklesForLoss?: number;//TFL
    passesDefended?: number;//PD
    quarterbackHits?: number;//QB HTS
    defensiveTouchdowns?: number;//TD
    //Fumbles
    totalFumbles?: number;//FUM
    fumblesLost?: number;//LST
    fumblesRecovered?: number;//REC
    //Interceptions
    interceptions?: number;//INT
    interceptionYards?: number;//YDS
    interceptionTouchdowns?: number;//TD
    //Kicking
    numberOfFieldGoals?: number;//FG
    fieldGoalPercentage?: number;//PCT
    longestFieldGoal?: number;//LONG
    extraPoints?: number;//XP
    extraPointAttempts?: number;//XPA
    points?: number;//PTS
    //Kick Returns
    numberOfKickReturns?: number;//NO
    totalKickReturnYards?: number;//YDS
    numberOfYardsPerKickReturn?: number;//AVG
    longestKickReturnYards?: number;//LONG
    kickReturnTouchdowns?: number;//TD
    //Passing
    completions?: number;//CMP
    passingAttempts?: number;//ATT
    passingYards?: number;//YDS
    yardsPerPassAttempt?: number;//AVG
    passingTouchdowns?: number;//TD
    interceptionsThrown?: number;//INT
    numberOfTimesSacked?: number;//SACK
    sackedYardsLost?: number;//SYD
    passerRating?: number;//RTG
    adjustedQbr?: number;//QBR
    //Punting
    numberOfPunts?: number;//NO
    totalPuntYards?: number;//YDS
    puntYardsPerPunt?: number;//AVG
    touchbacks?: number;//TB
    in20?: number;//in 20
    longestPuntYards?: number;//LONG
    //Punt Returns
    numberOfPuntReturns?: number;//NO
    totalPuntReturnYards?: number;//YDS
    puntReturnYardsPerPuntReturn?: number;//AVG
    longestPuntReturnYards?: number;//LONG
    puntReturnTouchdowns?: number;//TD
    //Receiving
    totalReceptions?: number;//REC
    totalReceivingYards?: number;//YDS
    averageYardsPerReception?: number;//AVG
    receivingTouchdowns?: number;//TD
    longestReception?: number;//LNG
    receivingTargets?: number;//TGTS
    //Rushing
    rushingAttempts?: number;//CAR
    averages?: number;//AVG
    rushingTouchdowns?: number;//TD
    longestRun?: number;//LNG

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.game) {
                this.game = new Game(obj.game);
            }
            if(obj?.player) {
                this.player = new Player(obj.player);
            }
            if(obj?.totalTackles) {
                this.totalTackles = obj.totalTackles;
            }
            if(obj?.soloTackles) {
                this.soloTackles = obj.soloTackles;
            }
            if(obj?.totalSacks) {
                this.totalSacks = obj.totalSacks;
            }
            if(obj?.tacklesForLoss) {
                this.tacklesForLoss = obj.tacklesForLoss;
            }
            if(obj?.passesDefended) {
                this.passesDefended = obj.passesDefended;
            }
            if(obj?.quarterbackHits) {
                this.quarterbackHits = obj.quarterbackHits;
            }
            if(obj?.defensiveTouchdowns) {
                this.defensiveTouchdowns = obj.defensiveTouchdowns;
            }
            if(obj?.totalFumbles) {
                this.totalFumbles = obj.totalFumbles;
            }
            if(obj?.fumblesLost) {
                this.fumblesLost = obj.fumblesLost;
            }
            if(obj?.fumblesRecovered) {
                this.fumblesRecovered = obj.fumblesRecovered;
            }
            if(obj?.interceptions) {
                this.interceptions = obj.interceptions;
            }
            if(obj?.interceptionYards) {
                this.interceptionYards = obj.interceptionYards;
            }
            if(obj?.interceptionTouchdowns) {
                this.interceptionTouchdowns = obj.interceptionTouchdowns;
            }
            if(obj?.numberOfFieldGoals) {
                this.numberOfFieldGoals = obj.numberOfFieldGoals;
            }
            if(obj?.fieldGoalPercentage) {
                this.fieldGoalPercentage = obj.fieldGoalPercentage;
            }
            if(obj?.longestFieldGoal) {
                this.longestFieldGoal = obj.longestFieldGoal;
            }
            if(obj?.extraPoints) {
                this.extraPoints = obj.extraPoints;
            }
            if(obj?.extraPointAttempts) {
                this.extraPointAttempts = obj.extraPointAttempts;
            }
            if(obj?.points) {
                this.points = obj.points;
            }
            if(obj?.numberOfKickReturns) {
                this.numberOfKickReturns = obj.numberOfKickReturns;
            }
            if(obj?.totalKickReturnYards) {
                this.totalKickReturnYards = obj.totalKickReturnYards;
            }
            if(obj?.numberOfYardsPerKickReturn) {
                this.numberOfYardsPerKickReturn = obj.numberOfYardsPerKickReturn;
            }
            if(obj?.longestKickReturnYards) {
                this.longestKickReturnYards = obj.longestKickReturnYards;
            }
            if(obj?.kickReturnTouchdowns) {
                this.kickReturnTouchdowns = obj.kickReturnTouchdowns;
            }
            if(obj?.completions) {
                this.completions = obj.completions;
            }
            if(obj?.passingAttempts) {
                this.passingAttempts = obj.passingAttempts;
            }
            if(obj?.passingYards) {
                this.passingYards = obj.passingYards;
            }
            if(obj?.yardsPerPassAttempt) {
                this.yardsPerPassAttempt = obj.yardsPerPassAttempt;
            }
            if(obj?.passingTouchdowns) {
                this.passingTouchdowns = obj.passingTouchdowns;
            }
            if(obj?.interceptionsThrown) {
                this.interceptionsThrown = obj.interceptionsThrown;
            }
            if(obj?.numberOfTimesSacked) {
                this.numberOfTimesSacked = obj.numberOfTimesSacked;
            }
            if(obj?.sackedYardsLost) {
                this.sackedYardsLost = obj.sackedYardsLost;
            }
            if(obj?.passerRating) {
                this.passerRating = obj.passerRating;
            }
            if(obj?.adjustedQbr) {
                this.adjustedQbr = obj.adjustedQbr;
            }
            if(obj?.numberOfPunts) {
                this.numberOfPunts = obj.numberOfPunts;
            }
            if(obj?.totalPuntYards) {
                this.totalPuntYards = obj.totalPuntYards;
            }
            if(obj?.puntYardsPerPunt) {
                this.puntYardsPerPunt = obj.puntYardsPerPunt;
            }
            if(obj?.touchbacks) {
                this.touchbacks = obj.touchbacks;
            }
            if(obj?.in20) {
                this.in20 = obj.in20;
            }
            if(obj?.longestPuntYards) {
                this.longestPuntYards = obj.longestPuntYards;
            }
            if(obj?.numberOfPuntReturns) {
                this.numberOfPuntReturns = obj.numberOfPuntReturns;
            }
            if(obj?.totalPuntReturnYards) {
                this.totalPuntReturnYards = obj.totalPuntReturnYards;
            }
            if(obj?.puntReturnYardsPerPuntReturn) {
                this.puntReturnYardsPerPuntReturn = obj.puntReturnYardsPerPuntReturn;
            }
            if(obj?.longestPuntReturnYards) {
                this.longestPuntReturnYards = obj.longestPuntReturnYards;
            }
            if(obj?.puntReturnTouchdowns) {
                this.puntReturnTouchdowns = obj.puntReturnTouchdowns;
            }
            if(obj?.totalReceptions) {
                this.totalReceptions = obj.totalReceptions;
            }
            if(obj?.totalReceivingYards) {
                this.totalReceivingYards = obj.totalReceivingYards;
            }
            if(obj?.averageYardsPerReception) {
                this.averageYardsPerReception = obj.averageYardsPerReception;
            }
            if(obj?.receivingTouchdowns) {
                this.receivingTouchdowns = obj.receivingTouchdowns;
            }
            if(obj?.longestReception) {
                this.longestReception = obj.longestReception;
            }
            if(obj?.receivingTargets) {
                this.receivingTargets = obj.receivingTargets;
            }
            if(obj?.rushingAttempts) {
                this.rushingAttempts = obj.rushingAttempts;
            }
            if(obj?.averages) {
                this.averages = obj.averages;
            }
            if(obj?.rushingTouchdowns) {
                this.rushingTouchdowns = obj.rushingTouchdowns;
            }
            if(obj?.longestRun) {
                this.longestRun = obj.longestRun;
            }
        }
    }
}